prow/cmd/phaino/local.go

```diff
@@ -550,3 +546,24 @@ func (opts *options) convertJob(ctx context.Context, log *logrus.Entry, pj prowa
 	}
 	return fmt.Errorf("grace period expired, aborted: %v", ctx.Err())
 }
+
+func getTimeout(pj prowapi.ProwJob, opts *options) time.Duration {
+	if pj.Spec.DecorationConfig.Timeout.Duration > 0 {
+		return pj.Spec.DecorationConfig.Timeout.Duration
+	}
+	if opts.timeout > 0 {
+		return opts.timeout
+	}
+	return time.Duration(0)
+}
+
+func getMinimumGracePeriod(minimum time.Duration, pj prowapi.ProwJob, opts *options, log *logrus.Entry) time.Duration {
+	if pj.Spec.DecorationConfig.GracePeriod.Duration >= minimum {
+		return pj.Spec.DecorationConfig.GracePeriod.Duration
+	}
+	if opts.grace >= minimum {
```

Comment:
my feeling is if the user passed `--grace` as argument, they want to override the spec from pj, so maybe do this check first?